WebApp/Controllers/BrowserController.cs

```diff
@@ -68,30 +68,30 @@ public List<Endpoint> CreateEndpointList()
             try
             {
                 string supervisorId = Session["supervisorId"].ToString();
-                IEnumerable<ApplicationInfoApiModel> applications = RegistryService.ListApplications();
+                IEnumerable<EndpointInfoApiModel> endpoints = RegistryService.ListEndpoints();
+                IEnumerable<ApplicationInfoApiModel> applications = Session["Applications"] as IEnumerable<ApplicationInfoApiModel>;
 
                 if (applications != null)
                 {
                     foreach (var application in applications)
                     {
                         if (application.SupervisorId == supervisorId)
                         {
-                            ApplicationRegistrationApiModel applicationRecord = RegistryService.GetApplication(application.ApplicationId);
-
-                            foreach (var elem in applicationRecord.Endpoints)
+                            foreach (var endpoint in endpoints)
                             {
-                                EndpointInfoApiModel endpointModel = RegistryService.GetEndpoint(elem.Id);
-
-                                Endpoint endpointInfo = new Endpoint();
-                                endpointInfo.EndpointId = elem.Id;
-                                endpointInfo.EndpointUrl = elem.Endpoint.Url;
-                                endpointInfo.SecurityMode = elem.Endpoint.SecurityMode != null ? elem.Endpoint.SecurityMode.ToString() : string.Empty;
-                                endpointInfo.SecurityPolicy = elem.Endpoint.SecurityPolicy != null ? elem.Endpoint.SecurityPolicy.Remove(0, elem.Endpoint.SecurityPolicy.IndexOf('#') + 1) : string.Empty;
-                                endpointInfo.SecurityLevel = elem.SecurityLevel;
-                                endpointInfo.ApplicationId = application.ApplicationId;
-                                endpointInfo.ProductUri = application.ProductUri;
-                                endpointInfo.Activated = endpointModel.ActivationState == EndpointActivationState.Activated || endpointModel.ActivationState == EndpointActivationState.ActivatedAndConnected;
-                                endpointList.Add(endpointInfo);
+                                if (endpoint.ApplicationId == application.ApplicationId)
```

Comment:
I need  the check, because ListAllEndpointsAsync() gives the list of all the endpoints for all the applications and in the UI are displayed only  the endpoints related to applications that belong to the same supervisor